### PR TITLE
Pluribus Networks pn_dscp_map_pri_map module with Unit tests

### PR DESCRIPTION
**Pull request by rajaspachipulusu17**
##### SUMMARY

Point feature of dscp map pri map for Pluribus Networks Netvisor switches 

##### ISSUE TYPE

- New Module Pull Request

##### COMPONENT NAME
pn_dscp_map_pri_map.py module with unit tests.

##### ADDITIONAL INFORMATION
ansible 2.4.0.0
  config file = None
  configured module search path = [u'/root/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python2.7/dist-packages/ansible
  executable location = /usr/local/bin/ansible
  python version = 2.7.12 (default, Dec  4 2017, 14:50:18) [GCC 5.4.0 20160609]

```

### CODE REVIEW

**Comment by mattclay** [comment]
CI failure in unit tests: https://app.shippable.com/github/ansible/ansible/runs/99187/9/tests

File: test/units/modules/network/netvisor/test_pn_dscp_map_pri_map.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
+# Copyright: (c) 2018, Pluribus Networks
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+import json
+
+from units.compat.mock import patch
+from ansible.modules.network.netvisor import pn_dhcp_map_pri_map
```
Comment (mattclay) [inline test/units/modules/network/netvisor/test_pn_dscp_map_pri_map.py]: Typo, this looks like it should be `pn_dscp_map_pri_map` here and in several places below.

**Comment by rajaspachipulusu17** [comment]
@Akasurde can you merge this?

Thanks,
Rajas

File: lib/ansible/modules/network/netvisor/pn_dscp_map_pri_map.py [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+  - This module can be used to update priority mappings in tables.
+options:
+  pn_cliswitch:
+    description:
+      - Target switch to run the CLI on.
+    required: False
+    type: str
+  state:
+    description:
+      - State the action to perform. Use 'update' to modify
```
Comment (Akasurde) [inline lib/ansible/modules/network/netvisor/pn_dscp_map_pri_map.py]:
```suggestion
      - State the action to perform. Use C(update) to modify
```

Comment (Akasurde) [reply lib/ansible/modules/network/netvisor/pn_dscp_map_pri_map.py]: Apply this to rest of modules too.

**Review by Akasurde — CHANGES_REQUESTED** [review]
Rest LGTM

File: lib/ansible/modules/network/netvisor/pn_dscp_map_pri_map.py (hunk 2 above)
Comment (rajaspachipulusu17) [reply lib/ansible/modules/network/netvisor/pn_dscp_map_pri_map.py]: I have somehow missed it. I will apply to future modules. 

Thanks,
Rajas

**Comment by rajaspachipulusu17** [comment]
Can you merge now @Akasurde ?

File: test/units/modules/network/netvisor/test_pn_dscp_map_pri_map.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+                cli_cmd=cli
+            )
+        module.exit_json(**results)
+
+    def load_fixtures(self, commands=None, state=None, transport='cli'):
+        self.run_nvos_commands.side_effect = self.run_cli_patch
+        if state == 'update':
+            self.run_check_cli.return_value = True
+
+    def test_dhcp_map_pri_map_t1(self):
```
Comment (Akasurde) [inline test/units/modules/network/netvisor/test_pn_dscp_map_pri_map.py]:
```suggestion
    def test_dscp_map_pri_map_t1(self):
```

File: test/units/modules/network/netvisor/test_pn_dscp_map_pri_map.py [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+        set_module_args({'pn_cliswitch': 'sw01', 'pn_name': 'foo',
+                         'pn_pri': '0', 'pn_dsmap': '40', 'state': 'update'})
+        result = self.execute_module(changed=True, state='update')
+        expected_cmd = '/usr/bin/cli --quiet -e --no-login-prompt  switch sw01 dscp-map-pri-map-modify  pri 0 name foo dsmap 40'
+        self.assertEqual(result['cli_cmd'], expected_cmd)
+
+    def test_dhcp_map_pri_map_t2(self):
```
Comment (Akasurde) [inline test/units/modules/network/netvisor/test_pn_dscp_map_pri_map.py]:
```suggestion
    def test_dscp_map_pri_map_t2(self):
```

File: test/units/modules/network/netvisor/test_pn_dscp_map_pri_map.py [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+        set_module_args({'pn_cliswitch': 'sw01', 'pn_name': 'foo',
+                         'pn_pri': '1', 'pn_dsmap': '8,10,12,14', 'state': 'update'})
+        result = self.execute_module(changed=True, state='update')
+        expected_cmd = '/usr/bin/cli --quiet -e --no-login-prompt  switch sw01 dscp-map-pri-map-modify  pri 1 name foo dsmap 8,10,12,14'
+        self.assertEqual(result['cli_cmd'], expected_cmd)
+
+    def test_dhcp_map_pri_map_t3(self):
```
Comment (Akasurde) [inline test/units/modules/network/netvisor/test_pn_dscp_map_pri_map.py]:
```suggestion
    def test_dscp_map_pri_map_t3(self):
```

File: test/units/modules/network/netvisor/test_pn_dscp_map_pri_map.py [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+
+    def setUp(self):
+        self.mock_run_nvos_commands = patch('ansible.modules.network.netvisor.pn_dscp_map_pri_map.run_cli')
+        self.run_nvos_commands = self.mock_run_nvos_commands.start()
+
+        self.mock_run_check_cli = patch('ansible.modules.network.netvisor.pn_dscp_map_pri_map.check_cli')
+        self.run_check_cli = self.mock_run_check_cli.start()
+
+    def tearDown(self):
+        self.mock_run_nvos_commands.stop()
```
Comment (Akasurde) [inline test/units/modules/network/netvisor/test_pn_dscp_map_pri_map.py:27]:
```suggestion
        self.mock_run_nvos_commands.stop()
        self.mock_run_check_cli.stop()
```

**Review by Akasurde — CHANGES_REQUESTED** [review]

File: test/units/modules/network/netvisor/test_pn_dscp_map_pri_map.py (hunk 6 above)
Comment (rajaspachipulusu17) [reply test/units/modules/network/netvisor/test_pn_dscp_map_pri_map.py:27]: Committed it.

**Review by Akasurde — APPROVED** [review]

**Comment by rajaspachipulusu17** [comment]
@Akasurde can you merge this?

**Comment by rajaspachipulusu17** [comment]
@Akasurde can you merge this?

Thanks,
Rajas

**Comment by rajaspachipulusu17** [comment]
Can someone merge? its been quite some time now.

Thanks,
Rajas

**Comment by rajaspachipulusu17** [comment]
@Akasurde can you merge this?

**Pull request merged**